lib/src/backend/metadata.dart

```diff
@@ -193,6 +200,7 @@ class Metadata {
         _skip = skip,
         _verboseTrace = verboseTrace,
         _chainStackTraces = chainStackTraces,
+        _retry = retry,
```

Comment:
Add a [`RangeError.checkNotNegative()`](https://api.dartlang.org/stable/1.23.0/dart-core/RangeError/checkNotNegative.html) call to the body as well. Same for `new Metadata.parse()`.